core/frontend/src/render/RenderPlan.ts

```diff
@@ -6,18 +6,44 @@
  * @module Rendering
  */
 
-import { ClipVector, Point3d, Vector3d } from "@itwin/core-geometry";
 import {
-  AmbientOcclusion, AnalysisStyle, ClipStyle, ColorDef, Frustum, GlobeMode, HiddenLine, Hilite, LightSettings, MonochromeMode, Npc, RenderTexture,
-  ThematicDisplay, ViewFlags, WhiteOnWhiteReversalSettings,
+  AmbientOcclusion, AnalysisStyle, Atmosphere, ClipStyle, ColorDef, Frustum, GlobeMode, HiddenLine, Hilite, LightSettings, MonochromeMode, Npc,
+  RenderTexture, ThematicDisplay, ViewFlags, WhiteOnWhiteReversalSettings,
 } from "@itwin/core-common";
+import { ClipVector, Constant, Matrix3d, Point3d, Vector3d } from "@itwin/core-geometry";
 import { FlashSettings } from "../FlashSettings";
 import { Viewport } from "../Viewport";
+import { ViewState3d } from "../ViewState";
 
 const scratchPoint3a = new Point3d();
 const scratchPoint3b = new Point3d();
 const scratchPoint3c = new Point3d();
 
+/**
+ * @internal
+ */
+export class RenderPlanEllipsoid {
+  public readonly ellipsoidCenter: Point3d;
+  public readonly ellipsoidRotation: Matrix3d;
+  public readonly ellipsoidRadii: Point3d;
+
+  constructor(ellipsoidCenter: Point3d, ellipsoidRotation: Matrix3d, ellipsoidRadii: Point3d) {
+    this.ellipsoidCenter = ellipsoidCenter;
+    this.ellipsoidRotation = ellipsoidRotation;
+    this.ellipsoidRadii = ellipsoidRadii;
+  }
+
+  public equals(other: RenderPlanEllipsoid): boolean {
+    if (this.ellipsoidCenter !== other.ellipsoidCenter)
```

Comment:
You are comparing object identities where I think you want to use `almostEqual` methods.

Reply:
Thanks for catching that, fixed